### DISCUSSION TITLE
In rose stem trac.log output, what controls whether succeeded tasks are hidden?

### DISCUSSION THREAD

**Discussion by MichaelWhitall** [MetOffice/simulation-systems#250] [UM]
Hi folks,

At some point this year, UM rose stem started doing a really nice thing where the output trac.log file (which you have to paste into ticket summaries) wouldn't bother listing all the apps that succeeded.  This makes the trac.log output on UM tickets much quicker and easier to check at a glance; you only need to see and check whether tasks did NOT succeed :)

However, I find that occasionally (seemingly randomly) rose stem still lists all the tasks, including succeeded ones, in the trac.log file table.  Does anyone know how I can control this behaviour, and force rose stem to hide succeeded tasks in the trac.log file?

**Comment by jennyhickson** [comment]
Hi Mike,

I can't remember if there is an option to control it (someone else might be able to jump in on that), but the default control is that it depends on the groups you run. If you've run one of the "standard" groups such as developer etc then the succeeded tasks are hidden. If you run something more obscure then the default behaviour is still to print them all to make it easier to see what tasks those groups included. The list of "common groups" can be seen at the top of the script here: 

https://github.com/MetOffice/SimSys_Scripts/blob/a243e9dd890ecdbe0cacddd0a78053d3e7dfc540/suite_report.py#L125 

Hope that helps
Jenny

**Reply by jennyhickson** [reply]
It looks like the "verbosity" option is what you're after. Option 4 would print everything except housekeeping, gatekeeper and all successful tasks regardless of group.

**Reply by james-bruten-mo** [reply]
Yeah that's right, I have a feeling you access it by repeated `-q`'s

**Reply by MichaelWhitall** [reply]
Ah d'oh yep that explains it, thanks!  The problem is that I often run a couple of different groups in one go, e.g.
rose stem --group=developer --group=xc40_ga_comorph --group=spice_ga_comorph
to run developer plus the comorph tests...

**Reply by james-bruten-mo** [reply]
You can regenerate the trac.log once the suite is done by `python3 ~frum/SimSys_Scripts/suite_report.py -S <path/to/suite> -q`

**Reply by MichaelWhitall** [reply]
Awesome thanks James, that's done it nicely :)
